tests/test_sleep_block.py

```diff
@@ -0,0 +1,123 @@
+import time
+from unittest.mock import MagicMock, patch
+from nio.util.support.block_test_case import NIOBlockTestCase
+from nio.modules.persistence.default import Persistence
+from nio.modules.threading import Event
+from nio.common.signal.base import Signal
+from ..sleep_block import Sleep
+
+
+class SleepEvent(Sleep):
+
+    def __init__(self, event):
+        super().__init__()
+        self.event = event
+
+    def notify_signals(self, signals, output_id='default'):
+        super().notify_signals(signals, output_id)
+        self.event.set()
+
+
+class TestSleep(NIOBlockTestCase):
+
+    def get_test_modules(self):
+        return self.ServiceDefaultModules + ['persistence']
+
+    def get_module_config_persistence(self):
+        return {'persistence': 'default'}
+
+    def test_sleep_block(self):
+        e = Event()
+        blk = SleepEvent(e)
+        self.configure_block(blk, {
+            'log_level': 'DEBUG',
+            'interval': {'seconds': 0.1}
+        })
+        blk.start()
+        signals = [Signal({'name': 'signal1'}),
+                   Signal({'name': 'signal2'})]
+        start = time.time()
+        blk.process_signals(signals)
+        self.assert_num_signals_notified(0)
+        e.wait()
```

Comment:
Seems like we just want to make sure it slept before notifying, could we just do this to simplify?

```python
self.assertTrue(e.wait(0.5)) # make sure it doesn't take longer than 0.5 seconds
self.assert_num_signals_notified(2)
blk.stop()
```